### PR TITLE
Pass ansible-lint w/ moderate profile and some skipped rules

### PR DESCRIPTION
**Pull request by dato**
- 0d1832be660c5bc6b7a942a233ae17f4d38a9358 Use FQDN for with_filetree (community.general)
- 1a8313a409d43c37c230f16d22daf29f3b6383ea Fix yaml whitespace lints (colons, comments, indentation)
- 00577cb6d4ad181d5c5eac6f5eb7cc57ac9ba329 Fix lint: yaml[truthy]
- 92de9c3c0af47a7270b8b0b1af5267c38d9c1549 Fix lint: yaml[octal-values]
- 580b9ddafe01b159bfa43796327d8bd36433a191 Fix lint: yaml[line-length], add yamllint conf
- 19a501c76495e219a3e8012487f08467a7492095 Fix lint: yaml[quoted-strings]
- a8537e5267ae4beb09ae27c8bee52b264742da74 ansible-lint with moderate profile and some skipped rules (passing)
- 8d0793226dc18cb5d49350abf9926d206ac6779a ansible-lint workflow

-----

### CODE REVIEW

File: .yamllint.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  octal-values:
+    forbid-implicit-octal: true  # Required by ansible-lint
+    forbid-explicit-octal: true  # Required by ansible-lint
+  quoted-strings:
+    quote-type: single
+    required: only-when-needed
+    extra-required:
+      - ^[0-9:]+$  # IPv6 addresses containing only numbers 0-9 are misparsed
+    extra-allowed:
+      - ^\^   # regular expressions
```
Comment (dato) [inline .yamllint.yaml:23]: I'm open to dropping this.

I'd say I find quoted regexes more readable, but I guess it's just a matter of being used to it.

**Review by NightTsarina — APPROVED** [review]
LGTM, thanks!
Some very minor suggestions, the only important thing is the octal handling, but I am sending a PR in a few minutes

File: apache2/defaults/main.yaml [hunk 2]
```diff
@@ -12,7 +12,7 @@ apache2__sites_available_dir: '{{ apache2__serverroot }}/sites-available'
 apache2__sites_enabled_dir: '{{ apache2__serverroot }}/sites-enabled'
 
 # Permissions for configuration files.
-apache2__conffile_mode: 0644
+apache2__conffile_mode: '0644'
```
[the remark this answers is not in the document]
Comment (dato) [reply apache2/defaults/main.yaml:15]: Thank you. To ensure merge order, I've changed the base branch of this PR.

File: .yamllint.yaml (hunk 1 above)
Comment (dato) [reply .yamllint.yaml]: Tongue-in-cheek fix is best kind of fix: 2b51041e6484e9704f0b734e7f4ede39f04c2965

**Review by NightTsarina — APPROVED** [review]
LGTM!

**Pull request merged**